controlplane/pkg/nsm/nsm_heal_processor.go

```diff
@@ -193,56 +194,54 @@ func (p *healProcessor) healDstDown(ctx context.Context, cc *model.ClientConnect
 		logger.Infof("NSM_Heal(2.1) Remote NSE heal is done on source side")
 		return false
 	}
-
-	waitCtx, waitCancel := context.WithTimeout(ctx, p.properties.HealTimeout*3)
-	defer waitCancel()
-
 	logger.Infof("NSM_Heal(2.2) Starting DST Heal...")
-
 	// We are client NSMd, we need to try recover our connection srv.
-	endpointName := cc.Endpoint.GetNetworkServiceEndpoint().GetName()
-
 	// Wait for NSE not equal to down one, since we know it will be re-registered with new endpoint name.
-	if !p.waitNSE(waitCtx, cc, endpointName, cc.GetNetworkService(), p.nseIsNewAndAvailable) {
-		// Mark endpoint as ignored.
-		ctx = common.WithIgnoredEndpoints(ctx, map[string]*registry.NSERegistration{
-			endpointName: cc.Endpoint,
-		})
-	}
+	ctx = p.waitForNSEUpdateContext(ctx, cc.Endpoint, cc)
 	// Fallback to heal with choose of new NSE.
-	requestCtx, requestCancel := context.WithTimeout(ctx, p.properties.HealRequestTimeout)
-	defer requestCancel()
+	for attempt := 0; attempt < p.properties.HealRetryCount; attempt++ {
+		attemptSpan := spanhelper.FromContext(ctx, fmt.Sprintf("healing-attempt-%v", attempt))
+		requestCtx, requestCancel := context.WithTimeout(attemptSpan.Context(), p.properties.HealRequestTimeout)
+		defer requestCancel()
+		defer attemptSpan.Finish()
```

Comment:
Seems like `attemptSpan.Finish` will be called twice for each span except the span on last iteration

Reply:
It support multiple cases, otherwise it will show wrong timings.